timely_beliefs/tests/test_db_subclassing.py

```diff
@@ -0,0 +1,127 @@
+from datetime import datetime, timedelta
+
+from pytz import timezone
+from sqlalchemy import Column, Float, ForeignKey, Integer
+from sqlalchemy.ext.declarative import declared_attr
+from sqlalchemy.orm import backref, relationship
+
+from timely_beliefs import DBBeliefSource, DBSensor
+from timely_beliefs.beliefs.classes import TimedBeliefDBMixin
+from timely_beliefs.db_base import Base
+from timely_beliefs.sources.classes import BeliefSourceDBMixin
+from timely_beliefs.tests import session
+
+
+class RatedSource(DBBeliefSource):
+    """ Subclassing a belief source, adding a field."""
+
+    rating = Column(Float(), default=0)
+
+    def __init__(self, rating: float = None, **kwargs):
+        self.rating = rating
+        DBBeliefSource.__init__(self, **kwargs)
+
+    def __repr__(self):
+        return "<RatedSource name=%s rating=%d>" % (self.name, self.rating)
+
+
+def test_subclassing_source(db):
+    session.add(RatedSource(name="test_source", rating=5))
+    session.add(RatedSource(name="dummy"))
+
+    q = session.query(RatedSource)
+    print(q.all())
+    print(db.tables.keys())
+
+    assert session.query(RatedSource).filter(RatedSource.rating == 5).count() == 1
+
+    # We made one with default rating (0) and in conftest two are made in advance
+    assert session.query(RatedSource).filter(RatedSource.rating == 0).count() == 3
+
+
+class RatedSourceInCustomTable(Base, BeliefSourceDBMixin):
+    """ A custom db class for representing belief sources.
+    BeliefSource properties are added via the Mixin, so the Base is added by us.
+    Here, we can change the table name.
```

Comment:
Great. Please also add some of this information to the main user documentation.

Reply:
I've done so. Please take a look.